sample/src/main/java/org/aerogear/graphqlandroid/activities/MainActivity.kt

```diff
@@ -77,13 +102,91 @@ class MainActivity : AppCompatActivity() {
 
         pull_to_refresh.setOnRefreshListener {
             doYourUpdate()
-
+            pull_to_refresh.isRefreshing = false
         }
 
         recycler_view.layoutManager = LinearLayoutManager(this)
         recycler_view.adapter = taskAdapter
 
-        fabAdd.setOnClickListener {
+        //Used for in-memory mutations
+        buttonOffline.setOnClickListener {
+
+            Log.e(TAG, "buttonOffline clicked")
+            OfflineMutationSender()
+        }
+
+        //Used for persistent mutations
+        persistence.setOnClickListener {
+
+
+            Log.e(TAG, "in persistence")
+            val listOfMutations = dbDao.getAllMutations()
+            Log.e(TAG, "in persistence size of list : ${listOfMutations.size}")
+
+            listOfMutations.forEach { storedmutation ->
+
+                val responseClassName = storedmutation.responseClassName
+
+                val classReflected: Class<*> = Class.forName(responseClassName)
+
+                val constructor = classReflected.constructors.first()
+
+                val parameters = constructor.parameterTypes
+
+                val jsonValues = arrayListOf<Any>()
+
+                //Put all the json values into a list
+                val jsonObj = storedmutation.valuemap
+
+                val iter = jsonObj.keys()
+                iter.forEach { key ->
+                    jsonValues.add(jsonObj.get(key))
+                }
+
+                Log.e("jsonValuesList ", " ${jsonValues.size}")
+
+                jsonValues.forEach {
+                    Log.e("jsonValuesList : ", " $it")
+                }
+
+
+                parameters.forEachIndexed { index, clazz ->
+                    //                println(clazz.name)
+                    Log.e("parameters : ", " ${clazz.name}")
+                    if (clazz.name.equals("com.apollographql.apollo.api.Input")) {
```

Comment:
If `clazz is Input` is more syntactical here.

Reply:
Yeah, I tried the same. It was not working with that so had to try an alter to this.